tests/core/emscripten_scan_registers.cpp

```diff
@@ -0,0 +1,35 @@
+#include <set>
+#include <emscripten.h>
+#include <stdio.h>
+
+std::set<int> seenInts;
+
+void scan(void* x, void* y) {
+  printf("scan\n");
+  int* p = (int*)x;
+  int* q = (int*)y;
+  while (p < q) {
+    //printf("  %d\n", *p);
+    seenInts.insert(*p);
+    p++;
+  }
+}
+
+int main() {
+  int x = EM_ASM_INT({ return 314159 });
```

Comment:
Is this needed? Makes for a better test/example is it is more vanilla C++?

Reply:
This is useful as the compiler can't see the 314159 value :) (as the EM_ASM code will be in JS. So this guarantees that we scan registers and do not work thanks to compiler optimizations. (volatile might be another way to do this)